### PR TITLE
master: Add data mover to e2e cherrypick of #863

### LINKED ISSUE #930: Add datamover CI back into the master branch
relates to recent resolution of #905

### PR DESCRIPTION
**Pull request by hhpatel14**
Fixes #930

### CODE REVIEW

**Comment by kaovilai** [comment]
```
"restore phase is: WaitingForPluginOperations; expected: Completed
```

**Comment by kaovilai** [comment]
if further commits needed, also pin volsync subscription version channel
image

File: tests/e2e/backup_restore_suite_test.go [hunk 1]
```diff
@@ -235,10 +262,6 @@ var _ = Describe("AWS backup restore tests", func() {
 			restoreName := fmt.Sprintf("%s-%s", brCase.Name, restoreUid.String())
 
 			// install app
-			updateLastInstallTime()
-			log.Printf("Installing application for case %s", brCase.Name)
-			err = InstallApplication(dpaCR.Client, brCase.ApplicationTemplate)
-			Expect(err).ToNot(HaveOccurred())
```
Comment (kaovilai) [inline tests/e2e/backup_restore_suite_test.go:165]: This was a duplicate call to install application causing majority of the issues.

**Comment by kaovilai** [comment]
added notes in [slack](https://redhat-internal.slack.com/archives/C0144ECKUJ0/p1681268984019509)

**Comment by kaovilai** [comment]
Some were failing because volsync subscription flake

**Comment by kaovilai** [comment]
mongo flakes
/retest

**Comment by kaovilai** [comment]
volsync not installed flake
/retest

**Comment by kaovilai** [comment]
Bye bye 4.8 4.9
/override 4.8-operator-e2e-aws
/override 4.8-operator-e2e-gcp
/override 4.9-operator-e2e-aws
/override 4.9-operator-e2e-gcp

File: go.mod [hunk 2]
```diff
@@ -24,7 +24,11 @@ require (
 	sigs.k8s.io/controller-runtime v0.12.2
 )
 
-require github.com/google/go-cmp v0.5.9
+require (
+	github.com/backube/volsync v0.4.0
```
Comment (kaovilai) [inline go.mod:28]: not too important we can follow up to bump this up to 0.7 later. This is only used to add to scheme to list them if test fails.
Not using any struct fields.

**Comment by hhpatel14** [comment]
https://github.com/openshift/release/pull/38278

File: go.mod (hunk 2 above)
Comment (hhpatel14) [reply go.mod:28]: We don't have to change the volsync version. I removed ocp 4.8 and 4.9 from master.

**Review by sseago — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (kaovilai) [reply go.mod:28]: The volsync version problem is not the go.mod problem.. but the subscription channel in yaml file.

**Pull request merged**